pkg/operator/controllers/azurensg/azurensg_controller.go

```diff
@@ -0,0 +1,77 @@
+package azurensg
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	machinev1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	maoclient "github.com/openshift/machine-api-operator/pkg/generated/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+)
+
+//AzureNSGReconciler is the controller struct
+type AzureNSGReconciler struct {
+	arocli        aroclient.Interface
+	maocli        maoclient.Interface
+	kubernetescli kubernetes.Interface
+	log           *logrus.Entry
+}
+
+//NewReconciler creates a new Reconciler
+func NewReconciler(log *logrus.Entry, arocli aroclient.Interface, maocli maoclient.Interface, kubernetescli kubernetes.Interface) *AzureNSGReconciler {
+	return &AzureNSGReconciler{
+		arocli:        arocli,
+		maocli:        maocli,
+		kubernetescli: kubernetescli,
+		log:           log,
+	}
+}
+
+//Reconcile fixes the Network Security Groups
+func (r *AzureNSGReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !instance.Spec.Features.ReconcileNSGs {
+		//reconciling NSGs is disabled
+		return reconcile.Result{}, nil
+	}
+	err = r.reconcileSubnetNSG(ctx)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
```

Comment:
this could be simplified to the following
```suggestion
	return reconcile.Result{}, err
```

Reply:
Thanks, fixed.